File: helpdesk/helpdesk/report/first_response_time_for_tickets/first_response_time_for_tickets.py
# Copyright (c) 2013, DontManage and contributors
# For license information, please see license.txt

from __future__ import unicode_literals

import dontmanage


def execute(filters=None):
    columns = [
        {
            "fieldname": "creation_date",
            "label": "Date",
            "fieldtype": "Date",
            "width": 300,
        },
        {
            "fieldname": "first_response_time",
            "fieldtype": "Duration",
            "label": "First Response Time",
            "width": 300,
        },
    ]

    data = dontmanage.db.sql(
        """
		SELECT
			date(creation) as creation_date,
			avg(first_response_time) as avg_response_time
		FROM `tabHD Ticket`
		WHERE
			date(creation) between %s and %s
			and first_response_time > 0
		GROUP BY creation_date
		ORDER BY creation_date desc
	""",
        (filters.from_date, filters.to_date),
    )

    return columns, data

File: helpdesk/helpdesk/report/first_response_time_for_tickets/first_response_time_for_tickets.js
// Copyright (c) 2016, DontManage and contributors
// For license information, please see license.txt
/* eslint-disable */

dontmanage.query_reports["First Response Time for Tickets"] = {
  filters: [
    {
      fieldname: "from_date",
      label: __("From Date"),
      fieldtype: "Date",
      reqd: 1,
      default: dontmanage.datetime.add_days(dontmanage.datetime.nowdate(), -30),
    },
    {
      fieldname: "to_date",
      label: __("To Date"),
      fieldtype: "Date",
      reqd: 1,
      default: dontmanage.datetime.nowdate(),
    },
  ],
  get_chart_data: function (_columns, result) {
    return {
      data: {
        labels: result.map((d) => d.creation_date),
        datasets: [
          {
            name: "First Response Time",
            values: result.map((d) => d.first_response_time),
          },
        ],
      },
      type: "line",
      tooltipOptions: {
        formatTooltipY: (d) => {
          let duration_options = {
            hide_days: 0,
            hide_seconds: 0,
          };
          return dontmanage.utils.get_formatted_duration(d, duration_options);
        },
      },
    };
  },
};
